### ISSUE TITLE
feature suggestion: fast mouse movement

### ISSUE THREAD

**Issue by jovrstra** [tig/mcec#42]
Hello,
I was looking a long time for a program like this and will integrate it to control a PC from my Loxone server.

One feature suggestion: the projects https://github.com/rvaiya/warpd and https://github.com/EsportToys/TPMouse offer an interesting way of controlling the mouse cursor, particularly applicable when using an input that does not repeat well. I have not yet tried if these tools work alongside mcec, such functionality native in mcec would be useful. 

J.

**Comment by tig** [comment]
Not exactly sure the use-case you have in mind, but MCEC already supports controlling the mouse.

See https://tig.github.io/mcec/documentation.html#:~:text=s%0Ashiftup%3Actrl-,Mouse%20Commands,-With%20Mouse%20commands

**Comment by jovrstra** [comment]
Yes, and it even supports jumping to absolute locations. My use case is controlling the mouse from the app of my Loxone home automation. While this app theoretically repeats keypresses, it is often too slow which makes mouse-movement slow on high resolution displays. 
The two projects I linked have a grid-mode which helps to quickly jump with the mouse to locations.: rather than relying on key repeat to travel large distances, it can be achieved with a couple of key presses. It may be quite a specific use case though... :-) And maybe both apps can run next to each other, which would also be a solution - I haven't been able to try that...

edit: I can of course define more keys that make bigger jumps with the mouse to facilitate movement.
edit2: I just realized that the limits for absolute movement are independent of resolution...  that means I can achieve it by assigning buttons for absolute movement, keeping track of the absolute position values and changing them on subsequent presses (and reset them after some timeout). It would not have the visual aid as the tools I posted, but functionality wise it would work. (it would be more difficult to combine it with a remote though)

**Comment by jovrstra** [comment]
Small follow-up question, as I'm trying to use warpd through mcec... how come hotkeys do not trigger for some programs? 
I am running warpd alongside mcec, and while the hotkeys for warpd work when issued from the keyboard, they do not work when issued via mcec... Is it because mcec sends to the foreground window only? ( but e.g. the hotkey to launch the gamebar works when issued from mcec). Or is it related to e.g. the difference between left alt and right alt? (I cannot try with tpmouse for example as it uses left shift + right shift + g as a hotkey).

**Comment by tig** [comment]
Closing as resolved. Since this was filed, MCEC added \`mouse:mtp,x,y\` (absolute screen-pixel positioning, normalized across the virtual desktop — \`src/Commands/MouseCommand.cs\`) and \`mouse:drag,x1,y1,x2,y2[,...]\` (atomic press-move-release), on top of the \`mt\`/\`mtv\` absolute-move commands you'd already confirmed covered your Loxone use case in this thread. Between pixel-space absolute jumps and atomic drag, this should fully cover fast/precise positioning without relying on repeat-limited relative movement. Reopen if \`mtp\` doesn't meet the need for a specific panel scenario.

**Comment by jovrstra** [comment]
Thanks!
